### PR TITLE
chore(uve): fixed url change when create new language

### LINKED ISSUE #30030: UVE: Losing Page Reference when creating a language version with different URL
### Parent Issue

No parent issue identified.

### Problem Statement

When creating a new version of a page in Spanish from an existing English page, if the URL of the page is modified during the creation of the new version, it inadvertently changes the reference to the original page. This causes an error message when attempting to swap back to the original English version: "This URL does not exist (The page you are trying to access does not exist)." However, when re-entering the edit mode, the error does not reappear, indicating an inconsistent reference management.

### Steps to Reproduce

1. Open the dotCMS instance and create a page in English.
2. Create a new version of this page in Spanish, modifying the URL during the creation of the new version.
3. Attempt to swap from the Spanish version back to the original English version.
4. Observe the error message: "This URL does not exist (The page you are trying to access does not exist)."
5. Re-enter the edit mode and notice that the error does not reappear.

### Acceptance Criteria

1. Swapping between different language versions of a page should:
   - Allow the user to switch seamlessly without any error messages.
   - Maintain consistent references and behavior across all attempts, without requiring additional actions like re-entering edit mode.
   - Ensure that the correct version is displayed when switching languages, even if the URL of the new version is modified.

### dotCMS Version

The specific version where the defect was observed is not mentioned. Please confirm the version.

### Proposed Objective

User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

- Attached video demonstrating the issue where the reference to the original page is altered.

https://github.com/user-attachments/assets/3031e5ff-d8e6-4aff-a1ab-a7408400e5d8

### Assumptions & Initiation Needs

1. Confirm that the environment is set up to support multi-language content.
2. Ensure the content and versioning system is properly configured.

### Quality Assurance Notes & Workarounds

Currently, no workaround is required beyond re-entering edit mode, but this is not ideal for user experience.

### Sub-Tasks & Estimates

- [ ] Investigate the cause of the incorrect reference management when modifying the URL in a new language version.
- [ ] Identify and implement a fix to ensure references remain correct when switching between different language versions.
- [ ] Test the fix across all supported browsers: Chrome, Firefox, and Edge.

### PR DESCRIPTION
**Pull request by KevinDavilaDotCMS**
### Showcase

https://github.com/user-attachments/assets/45fff7cb-cda7-4887-969a-a8d1a5094754

This pull request introduces several changes to the `dot-ema-shell` component and related files to handle URL changes and improve navigation. The most important changes include adding a new event for URL changes, updating the component to handle this event, and modifying the test cases accordingly.

### Event Handling Improvements:

* [`core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts`](diffhunk://#diff-8b931d83eab16b451a32f81032bb3344306fac0012f1fe03c2ab6a508d63e283L14-R15): Added a new custom event `URL_IS_CHANGED` to the `NG_CUSTOM_EVENTS` enum.
* [`core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts`](diffhunk://#diff-677330662fea6dadc7e48fd8455ec2a6fe60d624c7ed1f01f0a3e985aacd05c6R168-R172): Added a case for `URL_IS_CHANGED` in the `DotEmaShellComponent` to handle URL change events and navigate to the new URL if necessary. [[1]](diffhunk://#diff-677330662fea6dadc7e48fd8455ec2a6fe60d624c7ed1f01f0a3e985aacd05c6R168-R172) [[2]](diffhunk://#diff-677330662fea6dadc7e48fd8455ec2a6fe60d624c7ed1f01f0a3e985aacd05c6L183-R192)

### Component Updates:

* [`core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts`](diffhunk://#diff-677330662fea6dadc7e48fd8455ec2a6fe60d624c7ed1f01f0a3e985aacd05c6L28-R28): Removed the unused `SafeUrlPipe` import and its corresponding entry in the `declarations` array. [[1]](diffhunk://#diff-677330662fea6dadc7e48fd8455ec2a6fe60d624c7ed1f01f0a3e985aacd05c6L28-R28) [[2]](diffhunk://#diff-677330662fea6dadc7e48fd8455ec2a6fe60d624c7ed1f01f0a3e985aacd05c6L75)

### Test Case Enhancements:

* [`core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts`](diffhunk://#diff-8843e3a4ce8c16e83408b1a6dcc3ad54eaddd17f8e986bbdb502e11bd4446ab4R1054-R1084): Added a test case to verify that the component triggers navigation when the URL path property changes.

### Additional Functionality:

* [`dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp`](diffhunk://#diff-30035fadab1873324da0061912beafb411367ed539e701868ffa1b5cd3875b21R726-R727): Added a new function `notifyUVEWhenURLisChanged` to dispatch a custom event when the URL changes. [[1]](diffhunk://#diff-30035fadab1873324da0061912beafb411367ed539e701868ffa1b5cd3875b21R726-R727) [[2]](diffhunk://#diff-30035fadab1873324da0061912beafb411367ed539e701868ffa1b5cd3875b21R755-R763)

This PR fixes: #30030

### CODE REVIEW

File: core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts [hunk 1]
```diff
@@ -11,7 +11,8 @@ export enum NG_CUSTOM_EVENTS {
     CANCEL_SAVING_MENU_ORDER = 'cancel-save-menu-order',
     OPEN_WIZARD = 'workflow-wizard',
     DIALOG_CLOSED = 'dialog-closed',
-    EDIT_CONTENTLET_UPDATED = 'edit-contentlet-data-updated'
+    EDIT_CONTENTLET_UPDATED = 'edit-contentlet-data-updated',
+    URL_IS_CHANGED = 'url-is-changed'
```
Comment (rjvelazco) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts:15]: Do we need a new event for this? Can't we check if the URL changes in the `EDIT_CONTENTLET_UPDATED` event?

**Review by oidacra — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp [hunk 2]
```diff
@@ -750,6 +752,15 @@
...
     }
 
+    function notifyUVEWhenURLisChanged(data) {
+        var customEvent = document.createEvent('CustomEvent');
+        customEvent.initCustomEvent('ng-event', false, false, {
+            name: 'url-is-changed',
+            payload: data
+        });
+        document.dispatchEvent(customEvent);
+    }
```
Comment (rjvelazco) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp:762]: We have the `save/save-page` event in this file. We can include the new `URL` as part of that payload.

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts (hunk 1 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts:15]: I can send the data in that event, but I didn't want to change how that event works elsewhere (now only sent a boolean, no the data object), so I preferred to create another event

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp (hunk 2 above)
Comment (KevinDavilaDotCMS) [reply dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp:762]: The event save/save-page is not triggered when change a property in the same "flow" event as create a language

File: core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts (hunk 1 above)
Comment (zJaaal) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/shared/enums.ts:15]: Check the NgEvents handler in Shell component, we send the URL on Save, we can do the same we do there, there's no need for another event.

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp (hunk 2 above)
Comment (zJaaal) [reply dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp:762]: We don't care about the changing unless we save/publish, at least for now. 

We should check this on save and I believe we are getting the URL on the payload of that event

**Review by zJaaal — CHANGES_REQUESTED** [review]

**Comment by KevinDavilaDotCMS** [comment]
This issue is related to editing properties from the create page dialog in another language.
This action does not trigger the "save-page" event, it only triggers the "edit-contentlet-data-updated" event. So we have no way to get the new path of the page.
So this new event is being used to get this path and replace it from UVE.
After talking with the team, it was decided to use a new event instead of modifying the existing one, since it is an isolated case and modifying the existing event can have a side effect

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp (hunk 2 above)
Comment (KevinDavilaDotCMS) [reply dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp:762]: Check [my comment](https://github.com/dotCMS/core/pull/30697#issuecomment-2486712535) below

**Review by rjvelazco — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp [hunk 3]
```diff
@@ -723,6 +723,8 @@
                             languageId
                         }
                     };
+
+                    notifyUVEWhenURLisChanged(data);
```
Comment (fmontes) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_js_inc.jsp:727]: How do we know exactly here that the url is changed?

**Review by fmontes — CHANGES_REQUESTED** [review]

**Pull request merged**